src/java.base/share/classes/javax/crypto/spec/HKDFParameterSpec.java

```diff
@@ -0,0 +1,493 @@
+         * hardware crypto device. This method appends to the existing list of
+         * values or creates a new list if there is none yet.
+         * <p>
+         * This supports the use-case where a label can be applied to the IKM
+         * but the actual value of the IKM is not yet available.
+         * <p>
+         * An implementation should concatenate the input key materials into a
+         * single value once all components are available.
+         *
+         * @param ikm
+         *     the input key material value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code ikm} is null
+         */
+        public Builder addIKM(byte[] ikm) {
+            if (ikm == null) {
+                throw new NullPointerException("ikm must not be null");
+            }
+            if (ikm.length != 0) {
+                return addIKM(new SecretKeySpec(ikm, "Generic"));
+            } else {
+                return this;
+            }
+        }
+
+        /**
+         * Adds a salt to the builder.
+         * <p>
+         * {@code addSalt} may be called when the salt value is to be assembled
+         * piece-meal or if part of the salt is to be supplied by a hardware
+         * crypto device. This method appends to the existing list of values or
+         * creates a new list if there is none yet.
+         * <p>
+         * An implementation should concatenate the salt into a single value
+         * once all components are available.
+         *
+         * @param salt
+         *     the salt value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code salt} is null
+         */
+        public Builder addSalt(SecretKey salt) {
+            if (salt != null) {
+                salts.add(salt);
+            } else {
+                throw new NullPointerException("salt must not be null");
+            }
+            return this;
+        }
+
+        /**
+         * Adds a salt to the builder.
+         * <p>
+         * {@code addSalt} may be called when the salt value is to be assembled
+         * piece-meal or if part of the salt is to be supplied by a hardware
+         * crypto device. This method appends to the existing list of values or
+         * creates a new list if there is none yet.
+         * <p>
+         * An implementation should concatenate the salt into a single value
+         * once all components are available.
+         *
+         * @param salt
+         *     the salt value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code salt} is null
+         */
+        public Builder addSalt(byte[] salt) {
+            if (salt == null) {
+                throw new NullPointerException(
+                    "salt must not be null");
+            }
```

Comment:
Why not use `Objects.requireNonNull(T, String)`? Same goes to all other `addXXX()` methods.

Reply:
Yes